traffic_control/clients/python/to_access/__init__.py

```diff
@@ -0,0 +1,627 @@
+	authenticating with Traffic Ops. This is distinct from an exit code of ``32`` in that the
+	*format* of the arguments was correct, but there was some problem with the *value*. For example,
+	passing ``https://test:`` to :option:`--to_url` will cause an exit code of ``2``, not ``32``.
+4
+	An HTTP client error occurred. The HTTP stack will be printed to stdout as indicated by other
+	options - meaning by default it will only print the response payload if one was given, but will
+	respect options like e.g. :option:`--request_payload` as well as
+	:option:`-p`/:option:`--pretty`.
+5
+	An HTTP server error occurred. The HTTP stack will be printed to stdout as indicated by other
+	options - meaning by default it will only print the response payload if one was given, but will
+	respect options like e.g. :option:`--request_payload` as well as
+	:option:`-p`/:option:`--pretty`.
+32
+	This is the error code emitted by Python's :mod:`argparse` module when the passed arguments
+	could not be parsed successfully.
+
+.. note:: The way exit codes ``4`` and ``5`` are implemented is by returning the status code of the
+	HTTP request divided by 100 whenever it is at least 400. This means that if the Traffic Ops
+	server ever started returning e.g. 700 status codes, the exit code of the script would be 7.
+
+
+Module Reference
+================
+
+"""
+import json
+import logging
+import os
+import sys
+import typing
+
+import requests
+
+from trafficops.restapi import LoginError, OperationError, InvalidJSONError
+from trafficops.tosession import TOSession
+
+l = logging.getLogger()
+l.disabled = True
+logging.basicConfig(level=logging.CRITICAL+1)
+
+#: The full path to a file used to store the user's Mojolicious athentication cookie (currently unused)
+COOKIEFILE = "" #os.path.expanduser(os.path.join("~", ".to-auth.cookie"))
+
+def set_cookie(cookie: str):
+	"""
+	Writes the passed cookie to the :data:`COOKIEFILE` file for later use.
+
+	.. warning::
+		Currently this function is never used, and it depends on the value of :data:`COOKIEFILE`,
+		which at the moment is being set to an empty string.
+	"""
+	with open(COOKIEFILE, "w") as f:
+		f.write(cookie)
+
+def output(r: requests.Response, pretty: bool, reqHeader: bool, respHeader: bool, reqPayload: bool,
+           indent: typing.Union[int, str] = '\t'):
+	"""
+	Prints the passed response object in a format consistent with the other parameters.
+
+	:param r: The :mod:`requests` response object being printed
+	:param pretty: If :const:`True`, attempt to pretty-print payloads as JSON
+	:param reqHeader: If :const:`True`, print request line and request headers
+	:param respHeader: If :const:`True`, print response line and response headers
+	:param reqPayload: If :const:`True`, print the request payload
+	:param indent: An optional number of spaces for pretty-printing indentation (default is the tab character)
+	"""
+	if reqHeader:
+		print(r.request.method, r.request.path_url, "HTTP/1.1")
```

Comment:
Since this is the client and should be usable with Python2 until Python2 support is removed, we should maintain Python2+3 compatibility until that time. So, stuff like the `print` function being imported from the __future__ module, not being able to use type annotations, etc., should be followed.

Reply:
I was sort of hoping to get away with this since it's not actually an extension on the underlying client code, but you're probably right.